tomcat/src/main/java/org/apache/catalina/SessionManager.java

```diff
@@ -1,12 +1,12 @@
 package org.apache.catalina;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
 
 public class SessionManager implements Manager {
 
     private static final SessionManager INSTANCE = new SessionManager();
-    private static final Map<String, Session> SESSIONS = new HashMap<>();
+    private static final ConcurrentMap<String, Session> SESSIONS = new ConcurrentHashMap<>();
```

Comment:
```suggestion
    private static final Map<String, Session> SESSIONS = new ConcurrentHashMap<>();
```
위 처럼 Map으로 받지 않고 ConcurrentMap으로 받은 이유가 있을까요?

Reply:
`SessionManager`가 항상 동시성을 보장해야하는 `Map`임을 코드에 나타내고 싶었습니다!
하지만 다른 Map 구현체로 교체하거나 기능을 확장하려면 이 부분도 수정되어야 하니 OCP 원칙에 위배될 거 같아 보이네요.
다시 Map으로 원복하겠습니다!

Reply:
좋습니다! `=`연산자 뒤에 이미 `ConcurrentHashMap`이 명시되어 있습니다. 이를 통해서, 동시성을 보장함을 보일 수 있다고 생각합니다~